modules: memfault: Add support for uploading modem traces to Memfault (#21252)
nrfconnect/sdk-nrf: doc/nrf/libraries/debug/memfault_ncs.rst

Alternatively, you can manually trigger the coredump upload by calling the :c:func:`memfault_zephyr_port_post_data` function.
You can use the :c:func:`memfault_coredump_has_valid_coredump` function to check whether a coredump is available.

Automatic sending of modem trace to Memfault
=============================================

.. memfault_coredump_send_start

To post modem traces to Memfault together with coredumps, set the :kconfig:option:`CONFIG_MEMFAULT_NCS_POST_MODEM_TRACE_ON_COREDUMP` Kconfig option to ``y``.
This option is only supported for nRF91 Series devices.

> The modem trace will be posted as a CDR (Custom Data Record) to Memfault and depends on the :ref:`nRF Modem Trace library <nrf_modem_lib_trace>` being configured to use a trace backend that persist traces across reboots.

Reviewer: ```suggestion
The modem trace will be posted as a CDR (Custom Data Record) to Memfault and depends on the :ref:`modem_trace_module` being configured to use a trace backend that persists traces across reboots.
```

Reviewer: fixes the doc build error.

> Depending on the account Memfault account type you use, different rate limits are imposed on the number of CDRs that can be posted.

Reviewer: ```suggestion
Depending on the Memfault account type you use, different rate limits are imposed on the number of CDRs that can be posted.
```

> To circumvent this limit for certain devices, Developer Mode can be enable for that device to be able to post more CDRs.

Reviewer: ```suggestion
To circumvent this limit for certain devices, you can enable Developer Mode for that device to post more CDRs.
```
